discovery-provider/alembic/trigger_sql/handle_challenge_disbursements.sql

```diff
@@ -0,0 +1,35 @@
+create or replace function handle_challenge_disbursement() returns trigger as $$
+declare
+  reward_manager_tx reward_manager_txs%ROWTYPE;
+begin
+
+  select * into reward_manager_tx from reward_manager_txs where reward_manager_txs.signature = new.signature limit 1;
+	
+  if reward_manager_tx is not null then
+	  -- create a notification for the challenge disbursement
+	  insert into notification
+		(slot, user_ids, timestamp, type, specifier, metadata)
+	  values
+		(
+			new.slot,
+			ARRAY [new.user_id],
+			reward_manager_tx.created_at,
+			'challenge_reward', 'challenge_reward:' || new.user_id || ':' || new.specifier,
+			('{ "specifier": "' || new.specifier || '",  "challenge_id": "' || new.challenge_id || '",  "amount": ' || new.amount ||  '}')::json
+		);
+  end if;
+  return null;
+
+exception
+  when others then return null;
```

Comment:
Is it common for this function to have an exception?

I think it makes sense to add this around `create trigger` to make it act like `create trigger if not exists`... but we'd probably want to know about an exception in the function body right?

Reply:
if it's to handle a duplicate row it's better to add to the end of the `insert` a `on conflict do nothing;`